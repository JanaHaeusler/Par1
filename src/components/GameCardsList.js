import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import GameCard from './GameCard'

GameCardsList.propTypes = {
    savedGameProfiles: PropTypes.array,
}

export default function GameCardsList({savedGameProfiles, onDelete}) {

    return(
    <GameCardsWrapper>
        {savedGameProfiles.map(({location, date, players, winner, shots, _id}) => 
          <GameCard key={_id} id={_id} onDelete={onDelete} location={location} date={date} players={players} winner={winner} shots={shots}/>
        )}
    </GameCardsWrapper>
    )
}

const GameCardsWrapper = styled.div`
  display: grid;
  gap: 10px;
`