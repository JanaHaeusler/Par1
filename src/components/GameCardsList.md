```js
<GameCardsList
  savedGameProfiles={[
    {
      location: 'Horner Racecourse',
      date: '11/04/2020',
      players: 'John, Jane',
      winner: 'Jane',
      shots: '89',
      _id: '1',
    },
    {
      location: 'City Park',
      date: '11/13/2020',
      players: 'Fritz, Hanni, Nanni',
      winner: 'Fritz',
      shots: '35',
      _id: '2',
    },
  ]}
  onDelete={() => {}}
/>
```
