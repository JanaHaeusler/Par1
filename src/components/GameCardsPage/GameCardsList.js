import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import GameCard from './GameCard'

GameCardsList.propTypes = {
    savedGameProfiles: PropTypes.object.isRequired,
    onDelete: PropTypes.func.isRequired,
    onEdit: PropTypes.func.isRequired,
    switchToSavedGamePage: PropTypes.func.isRequired,
}

export default function GameCardsList({savedGameProfiles, onDelete, onEdit, switchToSavedGamePage}) {

  return(
    <GameCardsWrapper>
        {savedGameProfiles.allIds.map((id) => {
          const {location, date, players, winner, shots, _id} = savedGameProfiles.byId[id]
          return <GameCard 
                    key={_id} 
                    id={_id} 
                    onDelete={onDelete} 
                    onEdit={onEdit} 
                    location={location} 
                    date={date} 
                    players={players} 
                    winner={winner} 
                    shots={shots}
                    switchToSavedGamePage={switchToSavedGamePage}/>
        })}
    </GameCardsWrapper>
  )
}

const GameCardsWrapper = styled.div`
  display: grid;
  gap: 10px;
`