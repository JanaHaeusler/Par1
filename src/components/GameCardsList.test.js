import GameCardsList from './GameCardsList'
import { render } from '@testing-library/react'

describe('GameCardsList', () => {
  
    const savedGameProfiles = [
        {
          location: 'Horner Racecourse',
          date: '2020-12-24',
          players: 'John, Jane',
          winner: 'Jane',
          shots: '89',
          _id: '1',
        },
        {
          location: 'City Park',
          date: '2020-01-01',
          players: 'Fritz, Hanni, Nanni',
          winner: 'Fritz',
          shots: '35',
          _id: '2',
        },
      ]
  
    it('renders correctly', () => {
        const onDeleteMock = jest.fn()
        const { container, getByText, getAllByRole } = render(<GameCardsList savedGameProfiles={savedGameProfiles} onDelete={onDeleteMock}/>)
        
        expect(container.firstChild).toMatchSnapshot()

        expect(getByText('Horner Racecourse')).toBeInTheDocument()
        expect(getByText('2020-12-24')).toBeInTheDocument()
        expect(getByText('John, Jane')).toBeInTheDocument()
        expect(getByText('Jane')).toBeInTheDocument()
        expect(getByText('89')).toBeInTheDocument()
        
        expect(getByText('City Park')).toBeInTheDocument()
        expect(getByText('2020-01-01')).toBeInTheDocument()
        expect(getByText('Fritz, Hanni, Nanni')).toBeInTheDocument()
        expect(getByText('Fritz')).toBeInTheDocument()
        expect(getByText('35')).toBeInTheDocument()

        const buttons = getAllByRole('button')
        expect(buttons).toHaveLength(2)
      })
})