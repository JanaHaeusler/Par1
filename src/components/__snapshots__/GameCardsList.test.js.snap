// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCardsList renders correctly 1`] = `
.c1 {
  padding: 10px;
  display: grid;
  grid-template-rows: repeat(3,1fr);
  grid-template-columns: repeat(3,1fr);
  grid-gap: 10px;
  justify-items: center;
  box-shadow: 0 0 10px var(--primary-medium);
  border-radius: 10px;
  background-color: var(--primary-medium);
  font-size: 0.9rem;
  text-align: center;
}

.c2 {
  grid-column: 1 / 3;
  justify-self: start;
}

.c3 {
  justify-self: end;
}

.c4 {
  margin: 0;
  place-items: center;
  font-weight: 550;
}

.c5 {
  grid-column: 1 / 2;
  grid-row: 3 / 4;
  place-items: center;
}

.c6 {
  grid-column: 2 / 3;
  grid-row: 3 / 4;
  place-items: center;
}

.c7 {
  place-items: center;
}

.c0 {
  display: grid;
  gap: 10px;
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <div
      class="c2"
    >
      Horner Racecourse
    </div>
    <div
      class="c3"
    >
      11/04/2020
    </div>
    <h4
      class="c4"
    >
      Player(s)
    </h4>
    <span
      class="c5"
    >
      John, Jane
    </span>
    <h4
      class="c4"
    >
      Winner(s)
    </h4>
    <span
      class="c6"
    >
      Jane
    </span>
    <h4
      class="c4"
    >
      Total Shots
    </h4>
    <span
      class="c7"
    >
      89
    </span>
  </section>
  <section
    class="c1"
  >
    <div
      class="c2"
    >
      City Park
    </div>
    <div
      class="c3"
    >
      11/13/2020
    </div>
    <h4
      class="c4"
    >
      Player(s)
    </h4>
    <span
      class="c5"
    >
      Fritz, Hanni, Nanni
    </span>
    <h4
      class="c4"
    >
      Winner(s)
    </h4>
    <span
      class="c6"
    >
      Fritz
    </span>
    <h4
      class="c4"
    >
      Total Shots
    </h4>
    <span
      class="c7"
    >
      35
    </span>
  </section>
</div>
`;
